SpatialGDK/Source/SpatialGDK/Public/Utils/InspectionColours.h

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#pragma once
+
+#include "Containers/UnrealString.h"
+#include "Math/Color.h"
+#include "Math/UnrealMathUtility.h"
+#include "Misc/Char.h"
+
+// Mimicking Inspector V2 colouring from platform/js/console/src/inspector-v2/styles/colors.ts
+
+namespace SpatialGDK
+{
+	namespace
+	{
+		const int32 MIN_HUE = 10;
+		const int32 MAX_HUE = 350;
+		const int32 MIN_SATURATION = 60;
+		const int32 MAX_SATURATION = 100;
+		const int32 MIN_LIGHTNESS = 25;
+		const int32 MAX_LIGHTNESS = 60;
+
+		double GenerateValueFromThresholds(int64 Hash, int32 Min, int32 Max)
+		{
+			return Hash % FMath::Abs(Max - Min) + Min;
+		}
```

Comment:
All the operations are integers, why isn't the hue an integer in this case? I suppose if you want to keep a generic hsl to rgb function on doubles as the one below, the call site of that is where the int-double conversion should happen for the hue that was generated from a threshold.

Reply:
you're saying this should return int and cast to double outside the `GenerateValueFromThresholds` call yea?

Reply:
That's what I'd prefer, the cast is probably implicit.